services/venice-controller/src/test/java/com/linkedin/venice/controller/TestVeniceParentHelixAdmin.java

```diff
@@ -1846,6 +1846,68 @@ public void testUpdateStoreNativeReplicationSourceFabric() {
         "Native replication source fabric does not match after updating the store!");
   }
 
+  @Test
+  public void testDisableHybridConfigWhenActiveActiveOrIncPushConfigIsEnabled() {
+    String storeName = Utils.getUniqueString("testUpdateStore");
+    Store store = TestUtils.createTestStore(storeName, "test", System.currentTimeMillis());
+
+    store.setHybridStoreConfig(
+        new HybridStoreConfigImpl(
+            1000,
+            100,
+            -1,
+            DataReplicationPolicy.NON_AGGREGATE,
+            BufferReplayPolicy.REWIND_FROM_EOP));
+    store.setActiveActiveReplicationEnabled(true);
+    store.setIncrementalPushEnabled(true);
+    store.setNativeReplicationEnabled(true);
+    store.setChunkingEnabled(true);
+    doReturn(store).when(internalAdmin).getStore(clusterName, storeName);
+
+    doReturn(CompletableFuture.completedFuture(new SimplePubSubProduceResultImpl(topicName, partitionId, 1, -1)))
+        .when(veniceWriter)
+        .put(any(), any(), anyInt());
+
+    when(zkClient.readData(zkMetadataNodePath, null)).thenReturn(null)
+        .thenReturn(AdminTopicMetadataAccessor.generateMetadataMap(1, -1, 1));
+
+    parentAdmin.initStorageCluster(clusterName);
+    // When user disable hybrid but also try to manually turn on A/A or Incremental Push, update operation should fail
+    // loudly.
+    Assert.assertThrows(
+        () -> parentAdmin.updateStore(
```

Comment:
nit: we can call out that's `VeniceHttpException` will be thrown but it's okay too